src/VisualStudio/Core/Impl/SolutionExplorer/AnalyzerItem/AnalyzerItemSource.cs

```diff
@@ -217,10 +217,12 @@ private ImmutableArray<AnalyzerReference> GetFilteredAnalyzers(IEnumerable<Analy
                 // Analyzer dependency:
                 // 1. Must be an Analyzer file reference (we don't understand other analyzer dependencies).
                 // 2. Mush have no diagnostic analyzers.
-                // 3. Must have non-null full path.
-                // 4. Must not have any assembly or analyzer load failures.
+                // 3. Must have no source generators.
+                // 4. Must have non-null full path.
+                // 5. Must not have any assembly or analyzer load failures.
                 if (analyzerReference is AnalyzerFileReference &&
                     analyzerReference.GetAnalyzers(project.Language).IsDefaultOrEmpty &&
+                    analyzerReference.GetGenerators().IsDefaultOrEmpty &&
```

Comment:
Seems like this filtering logic is only being applied for legacy csproj based projects. For SDK style projects, there seems to be no filtering at all (it even shows pure dependencies in the node). This fix itself looks fine, but we should open a new work item on project-system repo to replicate this filtering logic for SDK style projects.

Reply:
Tagging @drewnoakes - I am not sure if the code for what shows up under `Dependencies -> Analyzers` node in SDK style projects is present in Roslyn repo or project-system repo.

Reply:
Filed https://github.com/dotnet/project-system/issues/6723 to track. We may want to or have to refactor Roslyn a bit if you want to have a common method to call into.

Reply:
Thanks @mavasani. I commented on the .NET Project System issue.